File: represent/bench/java_flat/SpecGenBench_0318__SpecGenBench__common__IsPrefix__IsPrefix.java
public class IsPrefix {

    public boolean isPrefix (String pat, String txt) {
        int i = 0;

        while(i < pat.length()) {
            if(i >= txt.length() || pat.charAt(i) != txt.charAt(i))
                return false;
            i = i + 1;
        }

        return true;
    }

}

File: represent/bench/c_flat/SpecGenBench_0318__SpecGenBench__common__IsPrefix__IsPrefix.c
#include <string.h>

int isPrefix(const char * pat, const char * txt);

int isPrefix(const char * pat, const char * txt) {

        int i = 0;

        while(i < ((int)strlen(pat))) {
            if(i >= ((int)strlen(txt)) || pat[i) != txt[i))
                return 0;
            i = i + 1;
        }

        return 1;
}
